src/main/java/com/hotels/hcommon/hive/metastore/client/api/CloseableMetaStoreClient.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) 2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.hcommon.hive.metastore.client.api;
+
+import java.io.Closeable;
+
+import org.apache.hadoop.hive.metastore.IHMSHandler;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+
+import com.hotels.hcommon.ssh.Tunnelable;
+
+public interface CloseableMetaStoreClient extends IMetaStoreClient, IHMSHandler, Closeable, Tunnelable {
```

Comment:
After our discussion this morning I had a look at these interfaces and how they're being used in WaggleDance. I'm a bit concerned about the overlap between them and also the fact that Tunnelable is being added in here which the WaggleDance "version" of this (CloseableIHMSHandler) doesn't have. For now can you remove IHMSHandler from this? Let's get this working end to end with Circus Train and do a release and then look at WaggleDance.

Reply:
What about `Tunnelable`?  This forces every implementation to be Tunnelable surely that's something the implementation should decide to add if needed??

Reply:
OK, I will remove the `IHMSHandler` and `Tunnelable` extensions

Reply:
On second thoughts, I will leave the `Tunnelable` interface extension as it greatly simplifies the factories and suppliers, tunneling is only activated if you pass the client into a TunnellableFactory and can be ignored in situations when you don't need it. That interface also doesn't contain any methods so doesn't add any cruft to implementations